Run singularity

ArangoGutierrez opened the pull request:
**Description of the Pull Request (PR):**

Create a `run-singularity` script to enable the functionality 
```
$ ./lolcow
 ________________________________________
/ You will be winged by an anti-aircraft \
\ battery.                               /
 ----------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||

```


**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [x] I have tested this PR locally with a `make test`
- [x] This PR is against the project's master branch
- [x] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge


Attn: @singularityware-admin

ArangoGutierrez committed:
create run-singularity script

ArangoGutierrez committed:
install run-singularity script

jmstover commented on mlocal/frags/Makefile.stub:
```diff
@@ -79,6 +81,12 @@ $(starter_suid_INSTALL): $(starter)
 
+# Run-Singularity bin script
+$(dist_bin_SCRIPTS_INSTALL): $(dist_bin_SCRIPTS)
+	@echo " INSTALL" $@
```
*EDIT:* Ignore this ...

Change this to:
`@echo " INSTALL run-singularity"`

yhcote replied:
yeah this is correct.

jmstover reviewed (dismissed):
@yhcote should look over

ArangoGutierrez committed:
add run-singularity to the list of files to clean

yhcote committed:
Merge branch 'master' into run_singularity

jmstover commented on mlocal/frags/Makefile.stub:
```diff
@@ -34,9 +36,9 @@ abs_BUILDDIR := `readlink -f $(BUILDDIR)`
+INSTALLFILES := $(singularity_INSTALL) $(starter_INSTALL) $(starter_suid_INSTALL) $(sessiondir) $(config_INSTALL) $(dist_bin_SCRIPTS_INSTALL)
 
-CLEANFILES += $(libruntime) $(starter) $(singularity) $(go_BIN) $(go_OBJ)
+CLEANFILES += $(libruntime) $(starter) $(singularity) $(go_BIN) $(go_OBJ) $(dist_bin_SCRIPTS_INSTALL)
```
You shouldn't need this here... this should be for files under `$(BUILDDIR)` ... which you don't have in your case. What would happen is:  `make clean`  would delete the file from where it was under `$(PREFIX)`

yhcote replied:
Lol I am just trying to resolve conflicts for this PR before I can review it. But thanks for joining in. And even without having read the patch yet, I think you're right. Gimme another minute ;)

yhcote commented on scripts/run-singularity:
```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
```
Is this an original file coming from 2.X ?

jmstover replied:
Yes, this should be exactly what `bin/run-singularity` is in the 2.x branch

ArangoGutierrez replied:
+1

yhcote commented on scripts/run-singularity:
```diff
@@ -0,0 +1,29 @@
+
+exec /usr/bin/env singularity --quiet run "$@"
+
+exit 255
```
Not a big deal but the end of the file is not an expected unix text file ending.

yhcote reviewed (changes requested):
Just a few things here and there and what @jmstover already pointed out.

ArangoGutierrez committed:
don't set run_singularity file on make CLEANFILES

make clean would delete the file from where it was under $(PREFIX)

ArangoGutierrez committed:
run_singularity to end in new line

run_singularity file to end with expected unix text file ending
A sequence of zero or more non- <newline> characters plus a terminating <newline> character.

ArangoGutierrez committed:
Merge branch 'master' into run_singularity

GodloveD reviewed (commented):
lgtm

Thanks @ArangoGutierrez !

ArangoGutierrez committed:
Merge branch 'master' into run_singularity

yhcote reviewed (approved):
Sorry for holding up that one! LGTM
